### PR TITLE
Bloch vector methods

### PR DESCRIPTION
**Pull request by AGaliciaMartinez**
**Description**
Clean-up of the methods `add_vectors` and `plot_vectors`.
- raise ValueError when wrong values on inputs for vectors and colors.
- fixes issue #1916 
- colors argument was bugged as it was not being taken into account when plotting. I kept the colors argument behaving as it was first intended, although we may want to change it later to match the behavior of points.
- Added tests for new ValueErrors and for the colors argument.

**Related issues or PRs**
#1913.

**Changelog**
Clean-up of the methods `add_vectors` and `plot_vectors`.
Now add_vectors raises ValueErrors for wrong vectors and colors argument
Fix issue #1916 
Fix colors argument
Improve test coverage

### CODE REVIEW

File: qutip/bloch.py [hunk 1]
```diff
@@ -415,29 +415,33 @@ def add_vectors(self, vectors, colors=None, alpha=1.0):
...
+        if vectors.ndim != 2 or vectors.shape[1] != 3:
+            raise ValueError(
+                "The included vectors are not valid. Vectors must "
+                "be equivalent to a 2D array where the first "
+                "index represents the iteration over the vectors and the "
+                "second index represents the position in 3D of vector head.")
+
+        n_vectors = vectors.shape[0]
+        if colors is None:
+            colors = np.array([None]*n_vectors)
```
Comment (hodgestar) [inline qutip/bloch.py:432]:
```suggestion
            colors = np.array([None] * n_vectors)
```

Comment (hodgestar) [reply qutip/bloch.py:432]: Maybe we should allow colors to be a scalar and then expand that to all vectors?

**Review by hodgestar — CHANGES_REQUESTED** [review]
Thanks -- this looks much nicer!

I've always found the wrapping around of the colours a bit odd, but I guess it's useful for plotting small number of vectors and not having to make up the colours oneself. Certainly using the auto-colouring and explicit colouring together gives rather bizarre results, but I think that's fine. Either the person writing the code cares enough to specify their own colours, or not.

File: qutip/bloch.py (hunk 1 above)
Comment (AGaliciaMartinez) [reply qutip/bloch.py:432]: Yes definitely. If you do not mind though, I'd rather do that in another PR once I have a better idea of how I would like to implement the kw arguments for plotting. There are in fact some challenges with points using two different methods for plotting and some arguments being attributes in the bloch sphere. I will try to address all of that in a single PR. For the moment, this fixes a few things and it already adds some small features.

Comment (hodgestar) [reply qutip/bloch.py]: Sound good.

**Review by hodgestar — APPROVED** [review]

**Pull request merged**